python/lsst/ip/isr/linearize.py

```diff
@@ -127,32 +139,280 @@ def fromYaml(self, yamlObject):
         yamlObject : `dict`
             Dictionary containing detector and amplifier information.
         """
+        self.setMetadata(metadata=yamlObject.get('metadata', None))
         self._detectorName = yamlObject['detectorName']
         self._detectorSerial = yamlObject['detectorSerial']
+        self._detectorId = yamlObject['detectorId']
         self.populated = True
         self.override = True
 
-        for amp in yamlObject['amplifiers']:
-            ampName = amp['name']
-            self.linearityCoeffs[ampName] = amp.get('linearityCoeffs', None)
+        for ampName in yamlObject['amplifiers']:
+            amp = yamlObject['amplifiers'][ampName]
+            self.linearityCoeffs[ampName] = np.array(amp.get('linearityCoeffs', None), dtype=np.float64)
             self.linearityType[ampName] = amp.get('linearityType', 'None')
             self.linearityBBox[ampName] = amp.get('linearityBBox', None)
 
+        if self.tableData is None:
+            self.tableData = yamlObject.get('tableData', None)
+            if self.tableData:
+                self.tableData = np.array(self.tableData)
+
+        return self
+
     def toDict(self):
         """Return linearity parameters as a dict.
 
         Returns
         -------
         outDict : `dict`:
         """
-        outDict = {'detectorName': self._detectorName,
+        # metadata copied from defects code
+        now = datetime.datetime.utcnow()
+        self.updateMetadata(date=now)
+
+        outDict = {'metadata': self.getMetadata(),
+                   'detectorName': self._detectorName,
                    'detectorSerial': self._detectorSerial,
-                   'hasTable': self._table is not None,
+                   'detectorId': self._detectorId,
+                   'hasTable': self.tableData is not None,
                    'amplifiers': dict()}
         for ampName in self.linearityType:
             outDict['amplifiers'][ampName] = {'linearityType': self.linearityType[ampName],
                                               'linearityCoeffs': self.linearityCoeffs[ampName],
                                               'linearityBBox': self.linearityBBox[ampName]}
+        if self.tableData is not None:
+            outDict['tableData'] = self.tableData.tolist()
+
+        return outDict
+
+    @classmethod
+    def readText(cls, filename):
+        """Read linearity from text file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file containing the linearity definition.
+        Returns
+        -------
+        linearity : `~lsst.ip.isr.linearize.Linearizer``
+            Linearity parameters.
+        """
+        data = ''
+        try:
+            with open(filename, 'r') as f:
+                data = yaml.load(f, Loader=yaml.FullLoader)
```

Comment:
Is the FullLoader needed because you are writing out numpy arrays natively into the YAML file (and I see PropertyList is not being turned into a dict first)? It still might be worth trying yaml.CFullLoader instead. It always gets a bit tricky to write special python classes to yaml.

Reply:
For now, I'm changing this to [[yaml.CLoader]], but there is an ongoing discussion in https://jira.lsstcorp.org/browse/DM-24225

Reply:
Good. Yes. That will help a lot and yes DM-24225 is where we are discussing the broader issue.